### PR TITLE
docs(sample): adding native-image-sample for java-trace

### PR DESCRIPTION
**Pull request by suztomo**
Importing google-cloud-trace's sample to this repository. I confirmed it works as `samples/native-image-sample/README.md`.

### CODE REVIEW

File: samples/native-image-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Cloud Trace Client Libraries with Native Image
+
+This application uses the Google Cloud [Trace Client Libraries](https://github.com/googleapis/java-trace) and can be compiled with Native Image.
+
+**Note:** In practice, the Trace Client Libraries are not used directly but through another tool, such as through the [OpenCensus Cloud Trace integration](https://cloud.google.com/trace/docs/setup/java) or through a framework like Spring via [Spring Cloud GCP Trace](https://github.com/spring-cloud/spring-cloud-gcp/blob/master/docs/src/main/asciidoc/trace.adoc).
+
+## Setup Instructions
+
+1. Follow the [GCP Project and Native Image Setup Instructions](../../README.md).
```
Comment (mpeddada1) [inline samples/native-image-sample/README.md:9]: This link will disappear once we archive the repo. How about we copy over the contents? https://github.com/GoogleCloudPlatform/native-image-support-java/blob/master/native-image-samples/README.md

File: samples/native-image-sample/pom.xml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <configuration>
+          <archive>
+            <manifest>
+              <mainClass>trace.TraceSampleApplication</mainClass>
+            </manifest>
+          </archive>
+        </configuration>
```
Comment (mpeddada1) [inline samples/native-image-sample/pom.xml:74]: could we modify this to:
```
     <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-jar-plugin</artifactId>
        <version>3.2.2</version>
        <configuration>
          <archive>
            <manifest>
              <addClasspath>true</addClasspath>
              <classpathPrefix>dependency-jars/</classpathPrefix>
              <mainClass>trace.TraceSampleApplication</mainClass>
            </manifest>
          </archive>
        </configuration>
      </plugin>
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-dependency-plugin</artifactId>
        <version>3.2.0</version>
        <executions>
          <execution>
            <id>copy-dependencies</id>
            <phase>package</phase>
            <goals>
              <goal>copy-dependencies</goal>
            </goals>
            <configuration>
              <outputDirectory>
                ${project.build.directory}/dependency-jars/
              </outputDirectory>
            </configuration>
          </execution>
        </executions>
      </plugin>
```
I realized that using just the maven-jar-plugin is not enough to create a working JAR. We also need to specify a classpath with the dependency JARs.

File: samples/native-image-sample/pom.xml [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            <artifactId>native-maven-plugin</artifactId>
+            <version>0.9.9</version>
+            <extensions>true</extensions>
+            <configuration>
+              <mainClass>trace.TraceSampleApplication</mainClass>
+              <buildArgs>
+                <buildArg>--no-fallback</buildArg>
+                <buildArg>--no-server</buildArg>
+                <buildArg>--initialize-at-build-time</buildArg>
+                <buildArg>--features=com.google.cloud.nativeimage.features.ProtobufMessageFeature</buildArg>
```
Comment (mpeddada1) [inline samples/native-image-sample/pom.xml:125]: What happens if we remove this `--features` option?

**Review by mpeddada1 — COMMENTED** [review]
Thank you! Added a few minor comments.

File: samples/native-image-sample/README.md (hunk 1 above)
Comment (suztomo) [reply samples/native-image-sample/README.md]: Good catch. Fixed.

File: samples/native-image-sample/pom.xml (hunk 2 above)
Comment (suztomo) [reply samples/native-image-sample/pom.xml:79]: That makes sense. I confirmed that `java -jar target/native-image-sample-0.1.0.jar` works.

File: samples/native-image-sample/pom.xml (hunk 3 above)
Comment (suztomo) [reply samples/native-image-sample/pom.xml:149]: It just works. Should I remove it?

Comment (mpeddada1) [reply samples/native-image-sample/pom.xml:149]: Cool! Yup, let's remove it. This sample doesn't need the extra protobuf configuration.

File: samples/native-image-sample/pom.xml (hunk 2 above)
Comment (mpeddada1) [reply samples/native-image-sample/pom.xml:79]: Nice! Thanks for checking.

File: samples/native-image-sample/pom.xml (hunk 3 above)
Comment (suztomo) [reply samples/native-image-sample/pom.xml]: Removed the argument.

**Comment by suztomo** [comment]
@mpeddada1 PTAL

**Review by mpeddada1 — APPROVED** [review]

**Pull request merged**